ditto-markov/ditto_markov/cache/cache.py

```diff
@@ -0,0 +1,62 @@
+from abc import ABC, abstractmethod
+from collections import OrderedDict
+import typing
+from typing import Union
+
+import cachetools
+
+
+class Cache(ABC):
+    @abstractmethod
+    def __getitem__(self, key):
+        raise NotImplementedError
+
+    @abstractmethod
+    def __setitem__(self, key, value):
+        raise NotImplementedError
+
+    @abstractmethod
+    def __contains__(self, key):
+        raise NotImplementedError
+
+
+RegisterableCache = Union[Cache, cachetools.Cache]
+
+
+class CacheRepository(Cache):
+    caches: typing.OrderedDict[str, RegisterableCache]
+
+    def __init__(self):
+        self.caches = OrderedDict()
```

Comment:
In recent versions of py dicts are ordered by default I'm pretty sure

Reply:
I started with an `OrderedDict`, realized that dicts are now ordered, and then decided to go back to `OrderedDict` just to make everything explicit.

Imagine having decades and decades of language behavior assumptions shattered